console-subscriber/tests/support/subscriber.rs

```diff
@@ -0,0 +1,323 @@
+                    .spawn(console_client(client_stream, test_state.clone()))
+                    .expect("console-test error: could not spawn 'console-client' task")
+                    .await
+                    .expect("console-test error: failed to await 'console-client' task");
+
+                test_state.advance_to_step(TestStep::UpdatesRecorded);
+                actual_tasks
+            })
+        })
+        .expect("console subscriber could not spawn thread");
+
+    tracing::subscriber::with_default(registry, || {
+        let runtime = tokio::runtime::Builder::new_current_thread()
+            .enable_all()
+            .build()
+            .unwrap();
+
+        runtime.block_on(async move {
+            test_state_test
+                .wait_for_step(TestStep::ClientConnected)
+                .await;
+
+            // Run the future that we are testing.
+            _ = task::Builder::new()
+                .name(MAIN_TASK_NAME)
+                .spawn(future)
+                .expect("console-test error: couldn't spawn test task")
+                .await;
+            _ = task::Builder::new()
+                .name(END_SIGNAL_TASK_NAME)
+                .spawn(futures::future::ready(()))
+                .expect("console-test error: couldn't spawn end signal task")
+                .await;
+            test_state_test.advance_to_step(TestStep::TestFinished);
+
+            test_state_test
+                .wait_for_step(TestStep::UpdatesRecorded)
+                .await;
+        });
+    });
+
+    let actual_tasks = join_handle
+        .join()
+        .expect("console-test error: failed to join 'console-subscriber' thread");
+
+    if let Err(test_failure) = validate_expected_tasks(expected_tasks, actual_tasks) {
+        panic!("Test failed: {test_failure}")
+    }
+}
+
+/// Starts the console server.
+///
+/// The server will start serving over its side of the duplex stream.
+///
+/// Once the server gets spawned into its task, the test state is advanced
+/// to the `ServerStarted` step. This function will then wait until the test
+/// state reaches the `UpdatesRecorded` step (indicating that all validation of the
+/// received updates has been completed) before dropping the aggregator.
+///
+/// # Test State
+///
+/// 1. Advances to: `ServerStarted`
+/// 2. Waits for: `UpdatesRecorded`
+async fn console_server(
+    server: console_subscriber::Server,
+    server_stream: DuplexStream,
+    mut test_state: TestState,
+) {
+    let ServerParts {
+        instrument_server: service,
+        aggregator,
+        ..
+    } = server.into_parts();
+    let aggregate = task::Builder::new()
+        .name("console::aggregate")
+        .spawn(aggregator.run())
+        .expect("client-console error: couldn't spawn aggregator");
+    Server::builder()
+        .add_service(service)
+        .serve_with_incoming(futures::stream::iter(vec![Ok::<_, std::io::Error>(
+            server_stream,
+        )]))
```

Comment:
very tiny, unimportant nit: could be simplified to just
```suggestion
        .serve_with_incoming(futures::stream::once(Ok::<_, std::io::Error>(server_stream)))
```